parsec/core/backend_connection/cmds.py

```diff
@@ -106,16 +106,19 @@ async def _send_cmd(transport: Transport, serializer, **req) -> dict:
         transport.logger.exception("Invalid response data", cmd=req["cmd"], error=exc)
         raise BackendProtocolError("Invalid response data") from exc
 
-    if rep["status"] == "invalid_msg_format":
-        transport.logger.error("Invalid request data according to backend", cmd=req["cmd"], rep=rep)
-        raise BackendProtocolError("Invalid request data according to backend")
-
-    if rep["status"] == "bad_timestamp":
-        raise BackendOutOfBallparkError(rep)
-
-    # Backward compatibility with older backends (<= v2.3)
-    if rep["status"] == "invalid_certification" and "timestamp" in rep["reason"]:
-        raise BackendOutOfBallparkError(rep)
+    if isinstance(rep, Dict):
```

Comment:
`isinstance(rep, dict)` (`Dict` is the typing thing so it's not Pythonic to use it here)

Reply:
also this mean we don't check for those in the case of block read/create which is an issue for `rep["status"] == "invalid_msg_format"`

And it will become more of an issue when we will migrate other schema and forget to take into account this part...

Instead we'd better do a
```python
if isinstance(rep, (BlockReadRep, BlockCreateRep)):
    # New Rust rep schemas
    status = rep.status
    reason = getattr(rep, "reason", "")
else:
    # Legacy Python rep schemas
    status = rep["status]
    reason = rep.get("reason", "")

if status == "invalid_msg_format":
    [...]

return rep
```

This way we ensure this place will break everytime a new schema is migrated, hence we must check this place en make sure it behavior is expected.